signature/sign.go

```diff
@@ -108,12 +108,12 @@ func Sign(key jwk.Key, sf SignedFielder, opts ...Option) (*pipeline.Signature, e
 	if pk, err := key.PublicKey(); err == nil && options.logger != nil {
 		fingerprint, err := pk.Thumbprint(crypto.SHA256)
 		if err != nil {
-			debug(options.logger, "Cannot calculate key thumbprint")
+			return nil, fmt.Errorf("calculating key thumbprint: %w", err)
 		} else {
 			debug(options.logger, "Public Key Thumbprint (sha256): %s", hex.EncodeToString(fingerprint))
 		}
-	} else {
-		debug(options.logger, "Unable to generate public key")
+	} else if err != nil {
+		return nil, fmt.Errorf("unable to generate public key: %w", err)
```

Comment:
I don't think this should be a fatal error, since it only impacts our debugging not the signing itself?

Reply:
I changed my mind, let's leave this as it was